presto-router/src/test/java/com/facebook/presto/router/TestScheduler.java

```diff
@@ -88,14 +107,33 @@ public void testWeightedRandomChoiceScheduler()
 
         HashMap<URI, Integer> hitCounter = new HashMap<>();
 
-        for (int i = 0; i < 1000; i++) {
+        for (int i = 0; i < 10_000; i++) {
             URI target = scheduler.getDestination("test").orElse(new URI("invalid"));
             assertTrue(servers.contains(target));
             assertTrue(weights.containsKey(target));
             hitCounter.put(target, hitCounter.getOrDefault(target, 0) + 1);
         }
 
-        assertTrue(hitCounter.get(new URI("192.168.0.3")) > hitCounter.get(new URI("192.168.0.1")));
+        //testing that ratios between weights roughly equate to ratios between actual destinations, may not pass 100% of the time
```

Comment:
```suggestion
        //testing that ratios between weights roughly equate to ratios between actual destinations
```

It needs to pass 100% of the time